.github/workflows/ci.yml

```diff
@@ -283,10 +283,8 @@ jobs:
     - setup
     - test-go
     - test-ui
-    if: |
-      always() &&
-      !contains(needs.*.result, 'failure') &&
-      !contains(needs.*.result, 'cancelled')
+    if: always()
     runs-on: ${{ fromJSON(needs.setup.outputs.compute-tiny) }}
     steps:
-    - run: echo "All Go tests successfully passed"
+    - run: |
+        grep -q -v -E '(failure|cancelled)' <<< "${{ needs.*.result }}"
```

Comment:
Looking at the GHA workflow run output, this is correct: https://github.com/hashicorp/vault/actions/runs/4598309610/jobs/8122612302#step:2:2

This should work: `"${{ toJSON(needs.*.result) }}"`

Reply:
Fixed, thanks!